### PR TITLE
feat(module_lexer): distinguish for types-only imports and exports

### PR DESCRIPTION
**Pull request by sxzz**

### CODE REVIEW

**Comment by Boshen** [comment]
image

😆

File: crates/oxc_module_lexer/src/lib.rs [hunk 1]
```diff
@@ -164,6 +171,8 @@ impl<'a> Visit<'a> for ModuleLexer<'a> {
             se: expr.span.end,
             d: ImportType::DynamicImport(expr.span.start + 6),
             a: expr.arguments.first().map(|e| e.span().start),
+            // TODO
```
Comment (Boshen) [inline crates/oxc_module_lexer/src/lib.rs:174]: Why is this todo?

Comment (sxzz) [reply crates/oxc_module_lexer/src/lib.rs]: Removed.

File: crates/oxc_module_lexer/src/lib.rs [hunk 2]
```diff
@@ -164,10 +171,37 @@ impl<'a> Visit<'a> for ModuleLexer<'a> {
...
+            if let TSType::TSLiteralType(literal) = &impt.parameter {
+                if let TSLiteral::StringLiteral(s) = &literal.literal {
+                    (Some(s.value.clone()), s.span.start, s.span.end)
+                } else {
+                    let span = literal.span();
+                    (None, span.start, span.end)
+                }
+            } else {
+                let span = impt.parameter.span();
+                (None, span.start, span.end)
```
Comment (sxzz) [inline crates/oxc_module_lexer/src/lib.rs:190]: Is there a better way to avoid repeating code here?

Comment (Boshen) [reply crates/oxc_module_lexer/src/lib.rs:190]: This is fine. What is being repeated?

**Comment by sxzz** [comment]
Made a tweak. PTAL

**Review by Boshen — APPROVED** [review]

**Comment by Boshen** [comment]
Thank you!

**Pull request merged**